src/filesystem.cc

```diff
@@ -948,28 +969,29 @@ ASFileSystem::ParsePath(
   return Status::Success;
 }
 
-ASFileSystem::ASFileSystem(const std::string& path) : as_regex_(AS_URL_PATTERN)
+ASFileSystem::ASFileSystem(
+    const std::string& path, const std::string& account_str,
+    const std::string& account_key)
+    : as_regex_(AS_URL_PATTERN)
 {
-  const char* account_str = std::getenv("AZURE_STORAGE_ACCOUNT");
-  const char* account_key = std::getenv("AZURE_STORAGE_KEY");
   std::shared_ptr<as::storage_account> account = nullptr;
   std::string host_name, container, blob_path, query;
   if (RE2::FullMatch(
           path, as_regex_, &host_name, &container, &blob_path, &query)) {
     size_t pos = host_name.rfind(".blob.core.windows.net");
     std::string account_name;
-    if (account_str == NULL) {
+    if (account_str.size()) {
       if (pos != std::string::npos) {
         account_name = host_name.substr(0, pos);
       } else {
         account_name = host_name;
       }
     } else {
-      account_name = std::string(account_str);
+      account_name = account_str;
     }
 
     std::shared_ptr<as::storage_credential> cred;
-    if (account_key != NULL) {
+    if (account_key.size()) {
       // Shared Key
```

Comment:
1. I would double check this part, not sure which logic is intended.
```cpp
// Before - these 2 conditions were opposites
if (account_str == NULL)
...
if (account_key != NULL)
...

// After - two conditions are the same?
if (account_str.size())
...
if (account_key.size())
...
```
2. Use `string.empty()` instead of `string.size()` so it's a bit more clear for the reader what the condition checks

Reply:
My bad. Thanks for catching. Corrected.